test/libdevcore/CommonData.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+	This file is part of solidity.
+ 	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+ 	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+ 	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*
+** Unit Tests for the StringUtils routines.
+*/
+
+#include <libdevcore/CommonData.h>
+#include <libdevcore/FixedHash.h>
+#include <test/Options.h>
+
+using namespace std;
+
+namespace dev
+{
+    namespace test
+    {
+        BOOST_AUTO_TEST_SUITE(CommonData)
+        BOOST_AUTO_TEST_CASE(test_format_number_readable)
+        {
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x8000000), "0x08 * 2**24");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x80000000), "0x80 * 2**24");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x800000000), "0x08 * 2**32");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x8000000000), "0x80 * 2**32");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x80000000000), "0x08 * 2**40");
+
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x7ffffff), "0x08 * 2**24 - 1");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x7fffffff), "0x80 * 2**24 - 1");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x7ffffffff), "0x08 * 2**32 - 1");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x7fffffffff), "0x80 * 2**32 - 1");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x7ffffffffff), "0x08 * 2**40 - 1");
+            BOOST_CHECK_EQUAL(formatNumberReadable((u256)0x88000000), "0x88 * 2**24");
```

Comment:
For small numbers like all the above I think it would make more sense displaying them in decimal.

Reply:
Say for <= 65536 it definitely makes sense.

Reply:
Okay great! I will get you these changes later today!~
Thanks for being quick and responsive!~

Reply:
Hey, just want to confirm here.
65536 = 0x10000

We were originally going with:
16777216 = 0x1000000

Would you like it at:
4294967296 = 0x100000000

instead? Im pushing up changes right now.

Reply:
Ah never mind, decimal would be nice if it would use a power of 10 on the right hand side.

Reply:
Could you clarify with an example output? 😺